### DISCUSSION TITLE
[json.exception.parse_error.101]

### DISCUSSION THREAD

**Discussion by KizzaGeorgeMbidde** [nlohmann/json#3582] [Q&A] [answered]
Hi.

The below is an error I get when I run the code given hereafter,

**terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 2, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
Aborted**

What is causing this error, and how can I resolve it?

Please note that this code an excerpt of the offending code, pulled out of a constructor of a class in my project.

Thank you.

`#include "json/json.hpp"

#include <vector>
#include <algorithm>
#include <iterator>
#include <fstream>

using nlohmann::json;

int main(int argc, char** argv)
{
    [[maybe_unused]] const auto i = argc;
    
    std::ifstream ifile{ argv[1] };
    std::vector<json> vjson;
    
    std::copy(std::istream_iterator<json>{ ifile },
                std::istream_iterator<json>{},
                std::back_inserter(vjson)
         );
    
    return 0;
}
`

**Comment by falbrechtskirchinger** [comment, marked as answer]
> **terminate called after throwing an instance of 'nlohmann::detail::parse_error' what(): [json.exception.parse_error.101] parse error at line 2, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal Aborted**
> 
> What is causing this error,

The last read from the stream fails because the stream has not yet reached the end but does so while reading a JSON value.

> and how can I resolve it?

You can catch the exception, but you won't be able to tell if parsing failed due to a syntax error or because the end of the stream was reached.

```cpp
try {
    std::copy(std::istream_iterator<json>{ifile},
              std::istream_iterator<json>{},
              std::back_inserter(vjson));
}
catch (const json::parse_error &) {}
```

Reading multiple JSON values from one input is not well supported and requires some work. I don't have a ready solution using a stream as input.

**Reply by KizzaGeorgeMbidde** [reply]
Thank you for your response, Florian.

However, I still get the below when I run the code,

**terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 2, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
Aborted**

**Reply by falbrechtskirchinger** [reply]
Then you must be having an issue somewhere else.

See https://godbolt.org/z/rj8sGxYq5 for a working example.

**Reply by nlohmann** [reply]
@KizzaGeorgeMbidde Do you need further assistance?

**Comment by KizzaGeorgeMbidde** [comment]
Hi.

Sorry for the delay in responding.

No, the help I received was sufficient. Originally, I was using the format of the entire constructor body being a try-catch block, as below,

`Constructor::Constructor()
try{

} catch(...){
}`

When I changed to the try-catch block within the constructor body, the above direction worked,

`Constructor::Constructor() { try{ } catch(...){ } }`

Thank you.
